### ISSUE TITLE
Error loading dff in MEd

### ISSUE THREAD

**Issue by Calvin-rgb** [Parik27/DragonFF#55]
Quite an old problem related to this add-on, but I decided to write about it on the gith hub page. Let it hang here.

**I did a little experiment:**
1. I created a simple cube model and applied a texture to it.
2. Added it to MEd:
![](https://imgur.com/KlsHZpw.png)
![](https://imgur.com/ZAKkgAI.png)

**Surprisingly, the model works well.**

After that, I created a model ('home.dff') with a large and complex mesh.
Bottom line: the model was not loaded into MEd
Screenshot for presentation:
![](https://imgur.com/CQAs8Di.png)

As I understand it, the problem is in the mesh, MEd reads simple meshes without problems.
In any case, you may be able to fix this problem even if it takes time.

**Comment by ThePortuguesePlayer** [comment]
DFFs from this addon also don't open in 3Ds using Kam's Tools.

**Comment by Parik27** [comment]
Just want to confirm - Were the export options for both the models the same? MEd and many other tools rely on the Material Split section to exist in the DFF. There's an option in the export settings to toggle that on or off. Ensure that that is checked before you export to make sure that the model loads in MEd.

**Comment by Calvin-rgb** [comment]
I plan to do a lot of research to find out what the error is.
If I find anything interesting, I will let you know here.

**Comment by Calvin-rgb** [comment]
I decided to use a third party gta tool for blender 2.79 (update script from ponz).
I opened the 'home.dff' model (not loaded in MEd) in 2.79 and exported it using this addon.
![](https://imgur.com/T3F3wXz.png)

All I did was put a check mark in the 'Write VCol' checkbox before exporting.
Screenshot from MEd:
![](https://imgur.com/AqT7C5W.png)


As you can see, the model loads perfectly, which cannot be said about the version from 2.9. If you have any ideas on this, I can throw off the download link for this add-on, maybe open source will help you succeed.

**Comment by Calvin-rgb** [comment]
More information regarding DragonFF:
I exported a regular cube labeled 'Export PLG' and it worked as it did before.
FURTHER I took my very beloved model of a house, which I exported more than once in GTA (it turned out only in blender 2.79 ponz), turned on the 'Export PLG' mark and as you already know, it did not work. I'm more inclined towards the PLG export problem, you will say that it works fine, but it seems strange to me that it works with a simple mesh (<100 polygons), as soon as you are going to export a HUGE model (several thousand polygons) it will not be able to do it ...

In the script from 'Ponz' 2.79, PLG export works on all models.

**Comment by Parik27** [comment]
I'll look into it. I mainly tested it with smaller meshes as well when I developed it, since the code should scale properly. If you could send a model that it doesn't work with, it'd be really helpful.

I'll also take a look at the KAMs incompatibility (although as I've said multiple times, that's usually an issue with the importer not supporting a bare-bones version of the dff, requiring additional information to import the dff that the game does not).

**Comment by Calvin-rgb** [comment]
Model: [home.zip](https://drive.google.com/file/d/18LYSrxC9FE_kXOoOFFLcVeLQSftlG7D1/view?usp=sharingl)

**Comment by Calvin-rgb** [comment]
If necessary, here is a script from Ponz with huge functionality of GTA tools, I think it will be useful for you to look and study it.
[Gta tools  2.79](https://drive.google.com/file/d/1KU7_vdFMv_N_y91xSApqlN5t6pKhZ4Ec/view?usp=sharing)

**Comment by Calvin-rgb** [comment]
I remember that I downloaded the 2dfx exporter for blender 2.79 from some github repository, and it worked very well. Only now I cannot find this repository now and I deleted the script itself :(

**Comment by Parik27** [comment]
Thanks for the model.

Yeah, I'm aware of the script from Ponz and have used it in the past. Unfortunately it is SA specific, so it's not gonna be tpp useful to look into. Furthermore, the code base for that plugin is significantly different, so I can't compare the two.

I remember someone mention the script as well, unfortunately I don't have it either. 2dfx support is something I plan to implement in - most of the code for it is already in the script, I was just looking for a good way to link it with the Blender interface so you could intuitively create/modify new 2d effects.

**Comment by Calvin-rgb** [comment]
It would be convenient if the tool opened in a separate window, as in 3ds max. Of course, this is very difficult to do, then just combine it into one tab in the sidebar.

**Comment by Parik27** [comment]
How would it be convenient? The current system is consistent with other plugins for Blender and is intuitive for the interface you'd find. 

I don't know how 3ds max opens in a separate window - do you mean the import/export options? Because I personally feel that the import/export menus in Blender are much better to work with. 

Combining everything into one sidebar also sounds like a bad idea and I don't think it's easy to do (or worth it for that matter). You don't need to see the mesh options when you're editing the materials (and vice-versa).

**Comment by ThePortuguesePlayer** [comment]
Personally, I'm not a fan of multiple windows, but I'm okay with multiple side panel tabs, specially if you can turn them on and off in the addon's settings menu.
2dfx would be very useful for me.

**Comment by Calvin-rgb** [comment]
I'm just used to 3d max, where the tool is in a separate window. But I agree with you.
